accepted/2022/version-selection.md

```diff
@@ -0,0 +1,447 @@
+
+- Roll-forward the SDK due to `global.json` pinning but retain runtime targeting as-is (as tested and expected).
+- Leave the SDK as-is (assume it works), but roll-forward the runtime from an end-of-life .NET version to a supported one.
+
+It is also reasonable to ask why we need `--roll-forward` switches for the CLI at all (for the SDK or the runtime). It is useful to specify a minimum version and roll forward policy (via the CLI or otherwise) when you want to enable your code to run in environments you don't control. This could be equally as true for scripts as apps.
+
+Note: The CLI doesn't offer a CLI affordance that matches `DOTNET_ROLL_FORWARD_TO_PRERELEASE`. We should wait for more feedback before adding pre-release-oriented switches to the CLI, for the runtime or SDK.
+
+### Host SDK selection
+
+The host offers `--fx-version` to select a specific runtime. We will add support to the same thing for selecting an SDK with `--sdk-version`.
+
+## Runtime version configuration
+
+Runtime version selection offers configuration via file, ENV, and CLI, detailed in [Runtime Binding Behavior](https://github.com/dotnet/designs/blob/main/accepted/2019/runtime-binding.md). This section describes the improvements that should be made to improve runtime version configuration.
+
+The following `*.runtimeconfig.json` file demonstrates the information needed to describe the version requirements for an ASP.NET Core application.
+
+```json
+{
+  "runtimeOptions": {
+    "tfm": "net6.0",
+    "frameworks": [
+      {
+        "name": "Microsoft.NETCore.App",
+        "version": "6.0.0"
+      },
+      {
+        "name": "Microsoft.AspNetCore.App",
+        "version": "6.0.0"
+      }
+    ],
+    "configProperties": {
+      "System.GC.Server": true,
+      "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization": false
+    }
+  }
+}
+```
+
+The key aspect to notice is that there are two runtimes listed, specified as "frameworks". In this case, the versions match exactly, but there is no requirement that they do so.
+
+Let's say you want to run this .NET 6 app on .NET 7 without source code changes. We can validate the efficacy of the existing configuration gestures to achieve that end.
+
+- `TargetFramework` -- Changing the `TargetFramework` MSBuild value to `net7.0` will result in the `frameworks` defaulting to a `7.0.0`.  That works, although requires a change to app source and for the app to be rebuilt.
+- `RuntimeFrameworkVersion` -- This MSBuild setting is an override on default `TargetFramework` behavior. It updates all frameworks to a single value, like `7.0.1`. This value can be specified in the project file or (more likely) as an argument to `dotnet build`. This option is one step better since it doesn't require updating source, but still requires a rebuild.
+- `version` in `runtimeconfig.json` -- This value can be hand-edited. It works and it is fine to do so, but it is a last resort and not considered a first-class option.
```

Comment:
Why is this not a first-class option for this scenario? It is super simple and everybody can pull it off. I think it is much easier to pull off than figuring out how to deploy and maintain a global versioning policy on enterprise desktops.